lib/action_subscriber/base.rb

```diff
@@ -61,17 +61,73 @@ def acknowledge
     end
 
     def _at_least_once_filter
+      processed_acknowledgement = false
       yield
-      acknowledge
+      processed_acknowledgement = acknowledge
     rescue => error
       ::ActionSubscriber::MessageRetry.redeliver_message_with_backoff(env)
-      acknowledge
+      processed_acknowledgement = acknowledge
       raise error
+    ensure
+      rejected_message = false
+      rejected_message = reject unless processed_acknowledgement
+
+      if !processed_acknowledgement && !rejected_message
+        Process.kill(:TTIN, Process.pid)
+        Process.kill(:USR2, Process.pid)
+
+        $stdout << <<-UNREJECTABLE
+          CANNOT ACKNOWLEDGE OR REJECT THE MESSAGE
+
+          This is a exceptional state for RabbitMQ to enter and puts the current
```

Comment:
I would `RabbitMQ` to `ActionSubscriber`. We haven't gotten rabbitmq into a weird state, its just our own internal state that is now inconsistent.